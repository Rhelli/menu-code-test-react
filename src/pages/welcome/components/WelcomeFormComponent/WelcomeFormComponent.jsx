import React, { useState } from 'react';
import DatePicker from 'react-date-picker';
import styles from './WelcomeFormComponent.module.scss';

const WelcomeFormComponent = () => {
  const [date, changeDate] = useState(new Date());
  const [description, setDescription] = useState(false);

  const handleDescriptionClick = () => {
    if (description === false) {
      setDescription(true);
    }
    if (description === true) {
      setDescription(false);
    }
    return description;
  };

  return (
    <div className={styles.welcomeFormComponent}>
      <h3>Make a booking</h3>
      <form className={styles.welcomeForm}>
        <div className={styles.partySize}>
          <p>Party Size</p>
          <select>
            <option value="2">For two</option>
          </select>
        </div>
        <div className={styles.timeDate}>
          <div>
            <p>Date</p>
            <DatePicker
              className={styles.datePicker}
              calendarClassName={styles.datePickerCalendar}
              onChange={changeDate}
              value={date}
              format={'dd-MM-y'}
              required={true}
            />
          </div>
          <div>
            <p>Time</p>
            <select />
          </div>
        </div>
        <button className={styles.formSubmitButton}>
          <p>Find A Table</p>
        </button>
      </form>
      <div className={description ? styles.descriptionFull : styles.description}>
        <p>The Chez Paree is the quintessential French dining experience, in the center of London&apos;s busy Soho.
        With a well earned Michelin star behind Jean-Luc Picard&apos;s Parisian gem, the <i>Chez Paree</i> is been
        critically acclaimed by numerous high-flyer critiques and restaurant-goers alike.</p>
        <button onClick={handleDescriptionClick}>+ Read More</button>
      </div>
    </div>
  );
};

export default WelcomeFormComponent;
