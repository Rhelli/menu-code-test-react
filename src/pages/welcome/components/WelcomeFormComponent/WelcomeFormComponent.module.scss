.welcomeFormComponent {
  width: 100%;
  height: auto;
  padding: 20px 0;

  h3 {
    width: 100%;
    text-align: center;
    font-size: 24px;
    color: #202020;
    border-bottom: 1px solid #d8d9db;
    padding-bottom: 14px;
  }

  .welcomeForm {
    display: flex;
    flex-flow: column nowrap;
    width: 100%;
    border-bottom: 1px solid #d8d9db;
    padding: 20px 0;

    .partySize,
    .timeDate {
      width: 100%;
      padding: 10px 0;

      p {
        font-size: 14px;
      }

      .datePicker {
        width: 100%;
        height: 30px;
        outline: none;
        border: none;
        border-bottom: 1px solid #d8d9db;

        div {
          width: 100%;
          border: 2px;
          
          button {
            &:hover,
            &:active,
            &:focus {
              svg {
                stroke: #da3743;
              }
            }
          }
        }

        &:hover {
          border-bottom: 2px solid #da3743;
        }
      }

      .datePickerCalendar {
        background-color: #F9EBEA;
        border-radius: 4px;
      }

      select {
        width: 100%;
        height: 30px;
        outline: none;
        border: none;
        border-bottom: 1px solid #d8d9db;
        background-color: transparent;

        &:hover {
          border-bottom: 2px solid #da3743;
        }
      }
    }

    .timeDate {
      display: flex;
      flex-flow: row nowrap;
      margin-bottom: 10px;

      div {
        width: 50%;

        &:first-child {
          margin-right: 6px;
        }
      }
    }

    .formSubmitButton {
      align-self: center;
      width: 55%;
      height: 45px;
      background-color: #da3743;
      border: none;
      outline: none;
      font-family: inherit;
      font-size: 16px;
      color: #fff;
      border-radius: 2px;
    }
  }

  .description,
  .descriptionFull {
    p {
      margin-top: 20px;
      width: 100%;
      line-height: 20px;
      color: #303030;
      overflow: hidden;
      white-space: nowrap;
      text-overflow: ellipsis;
    }

    button {
      outline: none;
      border: none;
      font-size: 12px;
      color: #da3743;
      background-color: transparent;

      &:hover {
        text-decoration: underline;
        cursor: pointer;
      }
    }
  }

  .description {
    p {
      height: 30px;
    }
  }

  .descriptionFull {
    p {
      height: -webkit-fill-available;
      overflow: unset;
      white-space: normal;
      text-overflow: unset;
    }
  }
}