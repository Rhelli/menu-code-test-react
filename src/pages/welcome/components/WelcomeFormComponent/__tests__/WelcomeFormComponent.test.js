import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import WelcomeFormComponent from '../WelcomeFormComponent';

Enzyme.configure({ adapter: new Adapter() });

const submitNewBooking = jest.fn(() => 'New Booking Submitted');
const welcomeFormComponent = (<WelcomeFormComponent submitNewBooking={submitNewBooking} />);

describe('The appearance and functionality of the WelcomeFormComponent', () => {
  const container = mount(<WelcomeFormComponent submitNewBooking={submitNewBooking} />);
  const form = container.find('.welcomeForm');

  it('Matches the previous snapshot', () => {
    const tree = renderer.create(welcomeFormComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    window.HTMLCanvasElement.prototype.getContext = () => {};
    mount(<WelcomeFormComponent submitNewBooking={submitNewBooking} />);
  });

  it('Should be contained by the master container with the correct attributes', () => {
    expect(container.childAt(0).name()).toBe('div');
    expect(container.childAt(0).hasClass('welcomeFormComponent')).toBe(true);
  });

  it('Contains the form element with the correct attributes', () => {
    expect(form.name()).toBe('form');
  });

  it('Submits the form with the correct function when the form is submitted inline', () => {
    form.simulate('submit');
    expect(submitNewBooking).toHaveBeenCalledTimes(1);
  });

  it('Contains inputs for party size, a date and a time', () => {
    expect(container.find('.partySize').childAt(1).name()).toBe('select');
    expect(container.find('.timeDate').childAt(0).childAt(1).name()).toBe('DatePicker');
    expect(container.find('.timeDate').childAt(1).childAt(1).name()).toBe('select');
  });
});
