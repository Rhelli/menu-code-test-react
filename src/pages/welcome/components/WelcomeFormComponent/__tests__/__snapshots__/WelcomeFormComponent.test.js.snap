// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the WelcomeFormComponent Matches the previous snapshot 1`] = `
<div
  className="welcomeFormComponent"
>
  <h3>
    Make a booking
  </h3>
  <form
    className="welcomeForm"
    onSubmit={[Function]}
  >
    <div
      className="partySize"
      onChange={[Function]}
    >
      <p>
        Party Size
      </p>
      <select
        id="partySize"
        name="partySize"
        required={true}
      >
        <option
          value="2"
        >
          For 2
        </option>
      </select>
    </div>
    <div
      className="timeDate"
    >
      <div>
        <p>
          Date
        </p>
        <div
          className="react-date-picker react-date-picker--closed react-date-picker--enabled datePicker"
          onFocus={[Function]}
        >
          <div
            className="react-date-picker__wrapper"
          >
            <div
              className="react-date-picker__inputGroup"
              onClick={[Function]}
            >
              <input
                max="275760-09-13"
                min="0001-01-01"
                name="date"
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                style={
                  Object {
                    "position": "absolute",
                    "visibility": "hidden",
                    "zIndex": "-999",
                  }
                }
                type="date"
                value="2021-06-10"
              />
              
              <input
                autoComplete="off"
                className="react-date-picker__inputGroup__input react-date-picker__inputGroup__day"
                data-input="true"
                inputMode="numeric"
                max={30}
                min={1}
                name="day"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                placeholder="--"
                required={true}
                type="number"
                value="10"
              />
              <span
                className="react-date-picker__inputGroup__divider"
              >
                -
              </span>
              <span
                className="react-date-picker__inputGroup__leadingZero"
              >
                0
              </span>
              <input
                autoComplete="off"
                autoFocus={false}
                className="react-date-picker__inputGroup__input react-date-picker__inputGroup__month react-date-picker__inputGroup__input--hasLeadingZero"
                data-input="true"
                inputMode="numeric"
                max={12}
                min={1}
                name="month"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                placeholder="--"
                required={true}
                type="number"
                value="6"
              />
              <span
                className="react-date-picker__inputGroup__divider"
              >
                -
              </span>
              <input
                autoComplete="off"
                autoFocus={false}
                className="react-date-picker__inputGroup__input react-date-picker__inputGroup__year"
                data-input="true"
                inputMode="numeric"
                max={275760}
                min={1}
                name="year"
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                placeholder="----"
                required={true}
                step={1}
                type="number"
                value="2021"
              />
              
            </div>
            <button
              className="react-date-picker__clear-button react-date-picker__button"
              onClick={[Function]}
              onFocus={[Function]}
              type="button"
            >
              <svg
                className="react-date-picker__clear-button__icon react-date-picker__button__icon"
                height={19}
                stroke="black"
                strokeWidth={2}
                viewBox="0 0 19 19"
                width={19}
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="4"
                  x2="15"
                  y1="4"
                  y2="15"
                />
                <line
                  x1="15"
                  x2="4"
                  y1="4"
                  y2="15"
                />
              </svg>
            </button>
            <button
              className="react-date-picker__calendar-button react-date-picker__button"
              onClick={[Function]}
              onFocus={[Function]}
              type="button"
            >
              <svg
                className="react-date-picker__calendar-button__icon react-date-picker__button__icon"
                height={19}
                stroke="black"
                strokeWidth={2}
                viewBox="0 0 19 19"
                width={19}
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="15"
                  width="15"
                  x="2"
                  y="2"
                />
                <line
                  x1="6"
                  x2="6"
                  y1="0"
                  y2="4"
                />
                <line
                  x1="13"
                  x2="13"
                  y1="0"
                  y2="4"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        onChange={[Function]}
      >
        <p>
          Time
        </p>
        <select
          id="time"
          name="time"
          required={true}
        >
          <option
            value="09:00am"
          >
            12:00pm
          </option>
          <option
            value="09:30am"
          >
            12:00pm
          </option>
          <option
            value="10:00am"
          >
            12:00pm
          </option>
          <option
            value="10:30am"
          >
            12:00pm
          </option>
          <option
            value="11:00am"
          >
            12:00pm
          </option>
          <option
            value="11:30am"
          >
            12:00pm
          </option>
          <option
            value="12:00pm"
          >
            12:00pm
          </option>
          <option
            value="12:30pm"
          >
            12:00pm
          </option>
          <option
            value="13:00pm"
          >
            12:00pm
          </option>
          <option
            value="13:30pm"
          >
            12:00pm
          </option>
          <option
            value="14:00pm"
          >
            12:00pm
          </option>
          <option
            value="14:30pm"
          >
            12:00pm
          </option>
          <option
            value="15:00pm"
          >
            12:00pm
          </option>
          <option
            value="15:30pm"
          >
            12:00pm
          </option>
          <option
            value="16:00pm"
          >
            12:00pm
          </option>
          <option
            value="16:30pm"
          >
            12:00pm
          </option>
          <option
            value="17:00pm"
          >
            12:00pm
          </option>
          <option
            value="17:30pm"
          >
            12:00pm
          </option>
          <option
            value="18:00pm"
          >
            12:00pm
          </option>
          <option
            value="18:30pm"
          >
            12:00pm
          </option>
          <option
            value="19:00pm"
          >
            12:00pm
          </option>
          <option
            value="19:30pm"
          >
            12:00pm
          </option>
          <option
            value="20:00pm"
          >
            12:00pm
          </option>
          <option
            value="20:30pm"
          >
            12:00pm
          </option>
          <option
            value="21:00pm"
          >
            12:00pm
          </option>
          <option
            value="21:30pm"
          >
            12:00pm
          </option>
          <option
            value="22:00pm"
          >
            12:00pm
          </option>
          <option
            value="22:30pm"
          >
            12:00pm
          </option>
        </select>
      </div>
    </div>
    <button
      className="formSubmitButton"
      type="submit"
    >
      <p>
        Continue
      </p>
    </button>
  </form>
</div>
`;
